Dev

Blake00789 opened the pull request.

Commit:
Tests CI

Blake00789 committed:
added dev branch

sch556 committed:
Added generic power ups and assets

Added a generic power up class, which when instantiated creates a power up that refills the current truck's water.

Added assets for blue, green and purple powerups.

Minor cleanup including muting music and resizing the game.

sch556 committed:
Added different power up types and updated to Java 8

5 different power up types; more damage, health refill, temporary
invincibility, increased speed, water refill. Adjusted firetruck,
entity, and gameobject classes to allow for this.

Updated to Java 8, so we can use Lambda functions and Java.time
libraries, allowing for power ups to only last a certain duration.

Commit:
Merge pull request #1 from Jordan00789/power_ups

Added generic power ups and assets

Blake00789 committed:
added save/load buttons.

Blake00789 committed:
basic save/load WIP

Blake00789 committed:
code so far loads trucks

sch556 committed:
Added random power up locations

sch556 committed:
WIP Adding loading capabilities

Also partially added unique identifiers (UUIDs) for each saveable object

jtt508 committed:
added easy, normal and hard modes

changes health and damage of the fortresses accordingly

sch556 committed:
Further progress on saving, tidy up

sch556 committed:
Improvements to loading

Fortresses now save and load properly

sch556 committed:
Powerup saving complete

sch556 committed:
Save/load functionality completed

Still need to add comments

sch556 committed:
Added comments to improve readability

Blake00789 committed:
Merge pull request #5 from Blake00789/gamemode

Added difficulty levels

Blake00789 committed:
Merge pull request #3 from Blake00789/power_ups

Added random power up locations

sch556 commented:
Merging errors mean that we don't have a lot of the save/load code, . I believe that this is also being merged into the wrong master branch; It should be merged into Blake00789:master not Luceapuce:master

sch556 committed:
Merge branch 'dev' into saves

sch556 committed:
Removed index from Fortress class

Blake00789 committed:
Merge pull request #4 from Blake00789/saves

Merge saves into the dev branch

sch556 committed:
Revert "Merge saves into the dev branch"

Blake00789 committed:
Merge pull request #6 from Blake00789/revert-4-saves

Revert "Merge saves into the dev branch"

sch556 committed:
Fixed issue with game running at 2x speed

Error was caused by bad automatic merging

sch556 committed:
Merge branch 'dev' into saves

sch556 committed:
Merge pull request #7 from Blake00789/saves

Fixed issue with game running at 2x speed

sch556 committed:
Revert "Merge branch 'dev' into saves"

This reverts commit 2a804304442648841a3a196603b310afd90af9ac, reversing
changes made to 1e3651126ef0160b6bd96781744f438f0155ed60.

sch556 committed:
Merge pull request #8 from Blake00789/saves

Revert "Merge branch 'dev' into saves"

sch556 committed:
Fixed performance issues

Issues caused by duplicate statbars, fixed by making them "static" in the game (not removed when 0).
Dead objects no longer remain in the dead objects array if not displayable, and are instead deleted.

sch556 committed:
Update Fortress.java

Fixed bug with healthbar not updating

sch556 committed:
Added basic testing for PowerUp class

sch556 committed:
Fixed bug of firetrucks not saving, and added runnable .jar file

sch556 committed:
Turned sound back on
